### PR TITLE
React bot framework secure

### PR DESCRIPTION
**Pull request by DingmaomaoBJTU**
|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                               |
| New feature?    | no                             |
| New sample?     | yes                               |
| Related issues? | fixes #X, partially #Y, mentioned in #Z |

## What's in this Pull Request?
Create a web part sample embeds the echo bot by using a webchat. As web part code is running on client side, [web chat security](https://blog.botframework.com/2018/09/01/using-webchat-with-azure-bot-services-authentication/) needs to be taken into consideration. This sample shows how to secure your conversation including:

- Use Direct Line token instead of Direct Line secret
- Tamper-proof user: for user id, generate it inside client side and detect if the client has changed the user ID and reject the change.

### CODE REVIEW

**Comment by hugoabernier** [comment]
Thank you @DingmaomaoBJTU, this is a fantastic sample! Thank you for taking the time to document the setup and deployment instructions so thoroughly.

Would you be interested in demoing this solution on our SPFx community call on December 3rd (3:00 PM GMT)? I think that a lot of people would like to see this solution.

Thanks again for your contribution!

**Comment by DingmaomaoBJTU** [comment]
Hi Hugo,

Thank you for your quick response. Sure we can have a 5 min demo at that time~ What kinds of demo do you prefer? Do you have any guide or requirement for this demo?

Thanks,
Qiong

**Comment by hugoabernier** [comment]
@DingmaomaoBJTU,

We usually allocate 15 minutes per demo, where we try to do the following:
- Discuss the business need (i.e.: why you would want to do this?)
- Demo the finished product
- Highlight key components of the code and/or deployment steps

You don't need to do a PowerPoint presentation, unless you want to. However, we do recommend using PowerPoint to highlight the key sections of your code (instead of scrolling around in the code editor) as it makes it easier for the audience to follow along (scrolling through a code editor tends to show up blurry on screen shares and makes it hard to follow). Alternatively, you can use the CodeTour Visual Studio Code extension and run the CodeTour during your presentation.

I'll confirm you on the agenda for December 3rd. I'm happy to help in any way I can. I can be reached on Twitter (@bernierh) or LinkedIn if you need anything.

Thank you again, and I'm really looking forward to your demo!

**Comment by hugoabernier** [comment]
As per #1590, I have you on the agenda for the December 3rd call. Looking forward to your demo.
